src/ledger/events.rs

```diff
@@ -0,0 +1,95 @@
+    pub publication_id: Uuid,
+
+    /// Datetime of the event publication.
+    ///
+    /// Format: ISO 8601
+    pub publication_datetime: DateTime<Utc>,
+
+    /// Idempotency key of the event payload (TBD: `account_address` + `transaction_hash`).
+    ///
+    /// It is unique for each distinct event, but consistent across retries or republications of the same event payload.
+    ///
+    /// Format: String (TBD)
+    pub idempotency_key: String,
+
+    /// Address of the account that is part of all account transfers.
+    ///
+    /// Format: Prefixed account address - 20 bytes - 0x1234567890123456789012345678901234567890
+    pub account_address: Address,
+
+    /// Hash of the Ethereum transaction that originated account transfers.
+    ///
+    /// Format: Prefixed transaction hash - 32 bytes - 0x1234567890123456789012345678901234567890123456789012345678901234
+    pub transaction_hash: Hash,
+
+    /// Address of the contract that originated account transfers.
+    ///
+    /// Format: Prefixed account address - 20 bytes - 0x1234567890123456789012345678901234567890
+    pub contract_address: Address,
+
+    /// Identifier of the Ethereum function that originated account transfers.
+    ///
+    /// Format: Prefixed function signature - 4 bytes - 0x12345678
+    pub function_id: [u8; 4],
+
+    /// Number of the block that originated account transfers.
+    ///
+    /// Format: Integer
+    pub block_number: BlockNumber,
+
+    /// Datetime of the block that originated account transfers.
+    ///
+    /// Format: ISO 8601
+    pub block_datetime: DateTime<Utc>,
+
+    /// List of account transfers that the related account_address is part of.
+    pub account_transfers: Vec<AccountTransfer>,
+}
+
+/// Represents a token transfer between a debit party and a credit party.
+pub struct AccountTransfer {
+    /// Address of the token contract that executed the transfer between `debit_party_address` and `credit_party_address`.
+    ///
+    /// It may differ from the main `contract_address` because any contract can execute transfers in token contracts.
+    ///
+    /// Format: Prefixed account address - 20 bytes - 0x1234567890123456789012345678901234567890
+    pub token_address: Address,
+
+    /// Address of the account from which the token was subtracted.
+    ///
+    /// Format: Prefixed account address - 20 bytes - 0x1234567890123456789012345678901234567890
+    pub debit_party_address: Address,
+
+    /// Address of the account to which the token was added.
+    ///
+    /// Format: Prefixed account address - 20 bytes - 0x1234567890123456789012345678901234567890
+    pub credit_party_address: Address,
+
+    /// Amount transferred from debit party to credit party.
+    ///
+    /// Format: TBD
+    /// * Prefixed hexadecimal amount - 32 bytes - 0x123450x1234567890123456789012345678901234567890678901234567890
+    /// * Decimal amount without prefix or padding - Integer - 1234
+    pub amount: U256,
```

Comment:
Deciding the format of the amount is critical. Having it as decimal can make math operations in queries a lot easier without having to perform a lot of conversions from hexadecimal.

Reply:
Agreed on decimal format.

Reply:
+1 for decimal format